constellation-app/constellation
# Add the ability to scroll to search term in conversation view

### Prerequisites


    for the "CI / Unit Tests") to complete first.

### Description of the Change

Added a feature to scroll to the message that matches a search term.  We are now able to scroll to each message that matches the given search term as well as having new buttons that allow you to go to either the previous match or the next match.  If you hit enter it will also go to the next match if the search term hasn't change so this is consistent with most other search features.
![image](https://user-images.githubusercontent.com/63685177/137887601-5e52cc6e-d48d-4daa-88a0-2366bb717798.png)

![image](https://user-images.githubusercontent.com/63685177/137887761-0b383c96-ebd2-4dfb-9128-de4da62de3d4.png)

An example showing the search, the previous and next buttons and iterating with ENTER in the text field.

![search_terms_conversation_view_1](https://user-images.githubusercontent.com/63685177/137888312-cc99c8f6-e57e-4324-a9b9-ac7234f96a55.gif)

### Alternate Designs

None considered

### Why Should This Be In Core?

This increases the usability of the conversation viewer allowing the user to easily find search terms in messages.  

### Benefits

Increases usability.  

### Possible Drawbacks



### Verification Process

1) Open graph that has content viewable in the conversation viewer
2) Open the conversation viewer
3) search for a term in the text field
4) You should now see the first message that matches the key term within the viewer
5) You should be able to see all of the matches by clicking the Previous and Next button.
6) In the search text field hit ENTER and the view should iterate through the messages with matches.  

### Applicable Issues

#958
